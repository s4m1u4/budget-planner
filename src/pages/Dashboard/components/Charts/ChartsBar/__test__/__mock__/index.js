export { TITLE, DATA } from "./mock";
