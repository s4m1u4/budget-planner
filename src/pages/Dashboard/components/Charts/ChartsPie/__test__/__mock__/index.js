export { TITLE, DATA_INCOME } from "./mock";
