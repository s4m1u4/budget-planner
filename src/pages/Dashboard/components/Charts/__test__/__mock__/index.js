export { CATEGORIES, RECORDS } from "./mock";
