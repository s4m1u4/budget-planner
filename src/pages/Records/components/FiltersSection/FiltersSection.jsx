import React from "react";
import { FiltersOptions } from "./FiltersOptions";
import { FiltersTypes } from "./FiltersTypes";
import { FiltersCategories } from "./FiltersCategories";

import {
  Section,
  SectionFooter,
  SectionHeader,
  SectionTitle,
} from "../../Records.styles";

export const FiltersSection = ({
  setPage,
  filters,
  setFilters,
  categories,
  getHistories,
  getFilteredHistories,
}) => {
  const handleClickSearch = async () => {
    await getFilteredHistories(filters.type, filters.category, 1);
    setPage(1);
  };

  const handleClickResetFilters = async () => {
    setFilters({ type: "", category: "" });
    await getHistories();
  };

  const handleChangeType = (event) => {
    setFilters((prevState) => ({
      ...prevState,
      type: event.target.value,
    }));
  };

  const handleChangeCategory = (event) => {
    setFilters((prevState) => ({
      ...prevState,
      category: event.target.value,
    }));
  };

  return (
    <Section>
      <SectionHeader>
        <SectionTitle>Filters</SectionTitle>
      </SectionHeader>
      <SectionFooter>
        <FiltersTypes
          filters={filters}
          categories={categories}
          handleChangeType={handleChangeType}
        />
        <FiltersCategories
          filters={filters}
          categories={categories}
          handleChangeCategory={handleChangeCategory}
        />
        <FiltersOptions
          filters={filters}
          handleClickSearch={handleClickSearch}
          handleClickResetFilters={handleClickResetFilters}
        />
      </SectionFooter>
    </Section>
  );
};
