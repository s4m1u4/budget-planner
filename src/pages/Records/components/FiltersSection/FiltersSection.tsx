import React, { ChangeEvent, FC } from "react";
import { FiltersOptions } from "./FiltersOptions";
import { FiltersTypes } from "./FiltersTypes";
import { FiltersCategories } from "./FiltersCategories";
import { IFilters } from "../../types";
import { ICategory } from "types";

import {
  Section,
  SectionFooter,
  SectionHeader,
  SectionTitle,
} from "../../Records.styles";

interface FiltersSectionProps {
  setPage: (page: number) => void;
  filters: IFilters;
  setFilters: ({ type, category }: IFilters | any) => void;
  categories: ICategory[];
  getHistories: (page?: number | null, limit?: number | null) => void;
  getFilteredHistories: (type: string, category: string, page: number) => void;
}

export const FiltersSection: FC<FiltersSectionProps> = ({
  setPage,
  filters,
  setFilters,
  categories,
  getHistories,
  getFilteredHistories,
}) => {
  const handleClickSearch = async () => {
    await getFilteredHistories(filters.type, filters.category, 1);
    setPage(1);
  };

  const handleClickResetFilters = async () => {
    setFilters({ type: "", category: "" });
    await getHistories();
    setPage(1);
  };

  const handleChangeType = (event: ChangeEvent<HTMLInputElement>) => {
    setFilters((prevState: IFilters) => ({
      ...prevState,
      type: event.target.value,
    }));
  };

  const handleChangeCategory = (event: ChangeEvent<HTMLInputElement>) => {
    setFilters((prevState: IFilters) => ({
      ...prevState,
      category: event.target.value,
    }));
  };

  return (
    <Section>
      <SectionHeader>
        <SectionTitle>Filters</SectionTitle>
      </SectionHeader>
      <SectionFooter>
        <FiltersTypes
          filters={filters}
          categories={categories}
          handleChangeType={handleChangeType}
        />
        <FiltersCategories
          filters={filters}
          categories={categories}
          handleChangeCategory={handleChangeCategory}
        />
        <FiltersOptions
          filters={filters}
          handleClickSearch={handleClickSearch}
          handleClickResetFilters={handleClickResetFilters}
        />
      </SectionFooter>
    </Section>
  );
};
