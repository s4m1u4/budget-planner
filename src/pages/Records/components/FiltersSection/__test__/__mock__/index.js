export { CATEGORIES, HISTORIES } from "./mock";
