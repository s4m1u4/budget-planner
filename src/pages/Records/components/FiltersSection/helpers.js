export const calculateRecordsByType = (records, filters) => {
  return records.filter((record) => record.type === filters.type);
};

export const calculateRecordsByCategories = (records, filters) => {
  const result = [];

  for (let i = 0; i < filters.categories.length; i++) {
    const filteredRecords = records.filter((record) => {
      return record.category.title === filters.categories[i];
    });

    if (!filteredRecords.length) {
      return [];
    }

    result.push(...filteredRecords);
  }

  return result;
};
