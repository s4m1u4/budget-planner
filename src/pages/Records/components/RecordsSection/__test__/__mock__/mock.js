export const RECORDS = [
  {
    id: "9cc6c0d7-24e3-46aa-aa7c-73e86c1dd230",
    amount: "2354.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "bc732adb-7b27-4d76-944c-4e7013eb97fa",
    amount: "836.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "9410db09-8e1c-4c76-a6b3-e766d9b53ae5",
    amount: "2454.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "e12eeb1b-661c-4c26-9f79-9c9f11f3441c",
    amount: "534.0000",
    type: "expense",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "d1afcb69-0b19-4ee3-85cb-e3a641df32e7",
    amount: "3454.0000",
    type: "expense",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "34b6368f-fc17-499f-9588-9cda11ec218d",
    amount: "1235.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "5861de79-061b-4f04-946c-ad2eaab28125",
    amount: "2364.0000",
    type: "expense",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "6b05bd63-623b-494f-91e0-5409ff9da167",
    amount: "3453.0000",
    type: "expense",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "2e3139b1-c463-4838-aecb-bb54e407d068",
    amount: "2642.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "6ac3498a-712b-4147-94d1-e3378c19e5b7",
    amount: "5676.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "6ac3498a-7dwb-4147-94d1-e3378c19e5b7",
    amount: "5676.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "6ac3498a-g4eb-4147-94d1-e3378c19e5b7",
    amount: "5676.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "6ac3498a-g3eb-4147-94d1-e3378c19e5b7",
    amount: "5676.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "6ac3498a-8geb-4147-94d1-e3378c19e5b7",
    amount: "5676.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "6ac3498a-97eb-4147-94d1-e3378c19e5b7",
    amount: "5676.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "6ac3498a-11eb-4147-94d1-e3378c19e5b7",
    amount: "5676.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
  {
    id: "6ac3498a-52eb-4147-94d1-e3378c19e5b7",
    amount: "5676.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
  },
];
