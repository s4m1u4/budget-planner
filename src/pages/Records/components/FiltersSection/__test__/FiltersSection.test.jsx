import { FiltersSection } from "../FiltersSection";
import { render, screen } from "@testing-library/react";
import { CATEGORIES } from "./__mock__";
import userEvent from "@testing-library/user-event";

const renderComponent = () => {
  return render(
    <FiltersSection
      setPage={setPage}
      filters={filters}
      setFilters={setFilters}
      categories={CATEGORIES}
      getHistories={getHistories}
      getFilteredHistories={getFilteredHistories}
    />
  );
};

const setPage = jest.fn();
const setFilters = jest.fn();
const getHistories = jest.fn();
const getFilteredHistories = jest.fn();
const filters = { type: "expense", category: "" };

describe("Filters section component", () => {
  it("snapshot matching", () => {
    const { baseElement } = renderComponent();
    expect(baseElement).toMatchSnapshot();
  });

  describe("should render", () => {
    beforeEach(() => {
      renderComponent();
    });

    it("a head", () => {
      expect(screen.getByText(/filters/i)).toBeInTheDocument();
    });

    it('a button "Type"', () => {
      expect(
        screen.getByRole("button", {
          name: /type/i,
        })
      ).toBeInTheDocument();
    });

    it('a button "Category"', () => {
      expect(
        screen.getByRole("button", {
          name: /category/i,
        })
      ).toBeInTheDocument();
    });

    it('a button "Search"', () => {
      expect(
        screen.getByRole("button", {
          name: /search/i,
        })
      ).toBeInTheDocument();
    });

    it('a button "Reset"', () => {
      expect(
        screen.getByRole("button", {
          name: /reset/i,
        })
      ).toBeInTheDocument();
    });
  });

  describe("should change", () => {
    beforeEach(() => {
      renderComponent();
    });

    it("type options", () => {
      userEvent.click(
        screen.getByRole("button", {
          name: /type/i,
        })
      );

      userEvent.click(screen.getByText(/income/i));
    });

    it("category options", () => {
      userEvent.click(
        screen.getByRole("button", {
          name: /category/i,
        })
      );

      userEvent.click(screen.getByText(/work/i));
    });
  });

  describe("should be", () => {
    beforeEach(() => {
      renderComponent();
    });

    it("search records", () => {
      userEvent.click(
        screen.getByRole("button", {
          name: /type/i,
        })
      );

      userEvent.click(screen.getByText(/income/i));

      userEvent.click(screen.getByRole("button", { name: /search/i }));
    });

    it("reset records", () => {
      userEvent.click(
        screen.getByRole("button", {
          name: /type/i,
        })
      );

      userEvent.click(screen.getByText(/income/i));

      userEvent.click(screen.getByRole("button", { name: /reset/i }));
    });
  });
});
