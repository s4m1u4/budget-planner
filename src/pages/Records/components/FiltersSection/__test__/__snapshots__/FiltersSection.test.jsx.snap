// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters section component snapshot matching 1`] = `
<body>
  <div>
    <div
      class="css-zeiouw"
    >
      <div
        class="css-1caytb6"
      >
        <h3
          class="css-1xqtn9i"
        >
          Filters
        </h3>
      </div>
      <div
        class="css-1gk4ta0"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-18gebku-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-8wl1e5-MuiButtonBase-root-MuiAccordionSummary-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Type
              </p>
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-1kh8flv-MuiAccordionDetails-root"
                  >
                    <fieldset
                      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                    >
                      <div
                        class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                        role="radiogroup"
                      >
                        <label
                          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                        >
                          <span
                            class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                          >
                            <input
                              class="PrivateSwitchBase-input css-1m9pwf3"
                              name="mui-1"
                              type="radio"
                              value="income"
                            />
                            <span
                              class="css-hyxlzm"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                data-testid="RadioButtonUncheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                                data-testid="RadioButtonCheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                          >
                            Income
                          </span>
                        </label>
                        <label
                          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                        >
                          <span
                            class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                          >
                            <input
                              checked=""
                              class="PrivateSwitchBase-input css-1m9pwf3"
                              name="mui-1"
                              type="radio"
                              value="expense"
                            />
                            <span
                              class="css-hyxlzm"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                data-testid="RadioButtonUncheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                                data-testid="RadioButtonCheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                          >
                            Expense
                          </span>
                        </label>
                      </div>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-18gebku-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-8wl1e5-MuiButtonBase-root-MuiAccordionSummary-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Category
              </p>
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-1kh8flv-MuiAccordionDetails-root"
                  >
                    <fieldset
                      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                    >
                      <div
                        class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                        role="radiogroup"
                      >
                        <label
                          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                        >
                          <span
                            class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                          >
                            <input
                              class="PrivateSwitchBase-input css-1m9pwf3"
                              name="mui-2"
                              type="radio"
                              value="084a3558-6b14-4e38-9d24-47e912021384"
                            />
                            <span
                              class="css-hyxlzm"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                data-testid="RadioButtonUncheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                                data-testid="RadioButtonCheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                          >
                            Work
                          </span>
                        </label>
                        <label
                          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                        >
                          <span
                            class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                          >
                            <input
                              class="PrivateSwitchBase-input css-1m9pwf3"
                              name="mui-2"
                              type="radio"
                              value="0ce314b9-a547-4f57-9d64-e466777ba8e9"
                            />
                            <span
                              class="css-hyxlzm"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                data-testid="RadioButtonUncheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                                data-testid="RadioButtonCheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                          >
                            Travel
                          </span>
                        </label>
                        <label
                          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                        >
                          <span
                            class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                          >
                            <input
                              class="PrivateSwitchBase-input css-1m9pwf3"
                              name="mui-2"
                              type="radio"
                              value="ea298d16-8bdb-4c5b-8198-6b2e5ce2f9a2"
                            />
                            <span
                              class="css-hyxlzm"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                                data-testid="RadioButtonUncheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                                data-testid="RadioButtonCheckedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                          >
                            Education
                          </span>
                        </label>
                      </div>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-p2xvpm"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-18gfzj-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Search
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-rk05t6-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Reset
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
